metagraph/src/graph/representation/canonical_dbg.cpp

```diff
@@ -125,13 +110,69 @@ void CanonicalDBG::map_to_nodes(std::string_view sequence,
     }
 }
 
+void CanonicalDBG
+::append_child_nodes_using_node_rev_comp(node_index node,
+                                         std::vector<node_index> &children) const {
+    /**
+     *
+     * find children of node by searching for parents of its reverse complement
+     * e.g., node = ATGGCT. Find TGGCTA and TGGCTT by looking for  TAGCCA and AAGCCA
+     *         TGGCTA      TAGCCA
+     *        /                  \
+     *  ATGGCT         ->         AGCCAT
+     *        \                  /
+     *         TGGCTT      AAGCCA
+     */
+
+    const auto &alphabet = graph_.alphabet();
+
+    //        rshift    rc
+    // ATGGCT -> TGGCT* -> *AGCCA
+    std::string rev_seq = get_node_sequence(node).substr(1) + std::string(1, '\0');
+    ::reverse_complement(rev_seq.begin(), rev_seq.end());
+    assert(rev_seq[0] == '\0');
+
+    // for each n, check for nAGCCA. If found, define and store the index for
+    // TGGCTrc(n) as index(nAGCCA) + offset_
+    const auto *dbg_succ = dynamic_cast<const DBGSuccinct*>(&graph_);
+    if (dbg_succ) {
+        const auto &boss = dbg_succ->get_boss();
+        dbg_succ->call_nodes_with_suffix_matching_longest_prefix(
+            std::string_view(&rev_seq[1], get_k() - 1),
+            [&](node_index next, uint64_t /* match length */) {
+                auto edge = dbg_succ->kmer_to_boss_index(next);
+                char c = boss.decode(boss.get_minus_k_value(edge, get_k() - 2).first);
+                if (c == boss::BOSS::kSentinel)
+                    return;
+
+                size_t i = boss.encode(::reverse_complement(c));
+
+                if (children[i] == DeBruijnGraph::npos)
+                    children[i] = next + offset_;
```

Comment:
KmerExtractorBOSS has methods `complement` and `reverse_complement`.
```suggestion
                boss::BOSS::TAlphabet c = boss.get_minus_k_value(edge, get_k() - 2).first;
                if (c == boss::BOSS::kSentinelCode)
                    return;

                c = KmerExtractorBOSS::complement(c);

                if (children[c] == DeBruijnGraph::npos)
                    children[c] = next + offset_;
```